src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -452,6 +453,11 @@ internal async Task HandleSyntaxParseRequest(SyntaxParseParams param, RequestCon
                 }
                 else
                 {
+                    if (await connectionService.TryRefreshAuthToken(scriptFile.ClientUri))
```

Comment:
it looks like you may be sending two SendResults per-request if the token needs to be refreshed since we don't return in the if or have the subsequent code in an else block.  This snippet may better look-up the default completion list since it would pass a null connectioninfo to GetCompletionItems method.

```C#

ConnectionInfo connInfo = null;
if (!await connectionService.TryRefreshAuthToken(scriptFile.ClientUri))
{
	// if token does not need to be refreshed then retrieve connection from connectionservice
	// get the current list of completion items and return to client
	ConnectionServiceInstance.TryFindConnection(
		scriptFile.ClientUri,
		out connInfo);
}
				
var completionItems = await GetCompletionItems(
	textDocumentPosition, scriptFile, connInfo);
await requestContext.SendResult(completionItems);
```

nit: the naming and semantics of `TryRefreshAuthToken` is not super-intuitive for me, since it succeeds when a refresh is needed, which seems to be more of the failure path in my mind.  Not a huge-deal if this makes more sense else in the code.